src/UKHO.ADDS.EFS.Domain/RetryPolicy/HttpClientPolicyProvider.cs

```diff
@@ -0,0 +1,172 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Polly;
+using UKHO.ADDS.Infrastructure.Results;
+
+namespace UKHO.ADDS.EFS.Domain.RetryPolicy
+{
+    /// <summary>
+    /// Provides Polly retry policies for HttpClient and custom result types to handle transient errors.
+    /// </summary>
+    public static class HttpClientPolicyProvider
```

Comment:
@harshal11869  should we rename this as this class's primary responsibility is to create and configure policy objects?
```suggestion
    public static class HttpRetryPolicyFactory
```

Reply:
Yes, you're right. Renamed to HttpRetryPolicyFactory.